assignment.rb

```diff
@@ -0,0 +1,310 @@
+            end
+            index += 1
+          end
+
+          # For easier access later on, let's identify which user we have verified.
+          user = @account_holders[ver_user]
+
+          # OK, the user has been validated and should be given the account menu.
+          if display_acc_menu == true
+            while display_acc_menu == true
+              acc_menu_sel = self.account_menu
+
+              # Verify the acc_menu_sel is a valid input, proceed only if able
+              if (1..3).include?(acc_menu_sel)
+                case acc_menu_sel
+                when 1
+                  # The user would like to view their available balance
+                  puts "Your current account balance is: $#{user.balance}"
+                  puts ""
+                when 2
+                  # The user would like to withdraw from their account
+                  print "How much would you like to withdraw? $"
+                  amount = gets.chomp.to_i
+                  puts ""
+                  # Let's make sure the account and the ATM can handle this transaction
+                  if user.can_withdraw?(amount) && @atm.can_withdraw?(amount)
+                    puts "Dispensing cash..." 
+                    user.deduct(amount)
+                    @atm.deduct(amount)
+                    puts "Your new balance is $#{user.balance}"
+                    puts "Thank you for paying that enourmous fee."
+                    puts ""
+                  elsif user.can_withdraw?(amount) == false
+                    puts "You have asked to withdraw more than you have in your account"
+                    puts ""
+                  elsif @atm.can_withdraw?(amount) == false
+                    puts "ATM has insufficient balance to fulfill this request."
+                    puts ""
+                  else
+                    puts "Something went wrong, and I don't know what"
+                    puts ""
+                  end
+                # The user would like to exit their account.  
+                else
+                  puts "Now exiting to the main menu."
+                  puts ""
+                  display_acc_menu = false
+                  main_men_sel_valid = false
+                end
+              # Invalid input in the account menu. Log out for security
+              else
+                puts "This was not a valid menu input."
+                puts "Logging you out for security purposes."
+                puts ""
+                display_acc_menu = false
+                main_men_sel_valid = false
+              end
+            end
+          # Login did not match records. 
+          else
+            puts "Your login does not match our records."
+            puts "Please try loggin in again."
+            puts ""
+            main_men_sel_valid = false
+          end
+          
+        when 2
+          # This code should view the atm's current balance
+          puts "This ATM currently has $#{@atm.balance}"
```

Comment:
This was so the user of the test program could check and actually see that my atm balance was dropping. You're right, this would be horrible on the real deal!